From: Peter Zijlstra <peterz@infradead.org>
Date: Wed, 26 Apr 2017 17:36:41 +0200
Subject: sched/topology: Remove FORCE_SD_OVERLAP
Git-commit: af85596c74de2fd9abb87501ae280038ac28a3f4
Patch-mainline: v4.13-rc1
References: bnc#1013018

Its an obsolete debug mechanism and future code wants to rely on
properties this undermines.

Namely, it would be good to assume that SD_OVERLAP domains have
children, but if we build the entire hierarchy with SD_OVERLAP this is
obviously false.

Signed-off-by: Peter Zijlstra (Intel) <peterz@infradead.org>
Cc: Linus Torvalds <torvalds@linux-foundation.org>
Cc: Mike Galbraith <efault@gmx.de>
Cc: Peter Zijlstra <peterz@infradead.org>
Cc: Thomas Gleixner <tglx@linutronix.de>
Cc: linux-kernel@vger.kernel.org
Signed-off-by: Ingo Molnar <mingo@kernel.org>
Signed-off-by: Mike Galbraith <mgalbraith@suse.de>
---
 kernel/sched.c          |    2 +-
 kernel/sched_features.h |    1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

--- a/kernel/sched.c
+++ b/kernel/sched.c
@@ -8046,7 +8046,7 @@ static int build_sched_domains(const str
 		sd = NULL;
 		for (tl = sched_domain_topology; tl->init; tl++) {
 			sd = build_sched_domain(tl, &d, cpu_map, attr, sd, i);
-			if (tl->flags & SDTL_OVERLAP || sched_feat(FORCE_SD_OVERLAP))
+			if (tl->flags & SDTL_OVERLAP)
 				sd->flags |= SD_OVERLAP;
 			if (cpumask_equal(cpu_map, sched_domain_span(sd)))
 				break;
--- a/kernel/sched_features.h
+++ b/kernel/sched_features.h
@@ -77,4 +77,3 @@ SCHED_FEAT(NONIRQ_POWER, 1)
  */
 SCHED_FEAT(TTWU_QUEUE, 1)
 
-SCHED_FEAT(FORCE_SD_OVERLAP, 0)
